### PR TITLE
fix: pass bearer token to curl using --header instead of --oauth2-bearer

### LINKED ISSUE #96161: Running ./hack/update-openapi-spec.sh leaves api/openapi-spec/swagger.json file empty
**What happened**:
After running ./hack/update-openapi-spec.sh with the latest code, api/openapi-spec/swagger.json file became empty even when no changes were made in the code.

**What you expected to happen**:
After running ./hack/update-openapi-spec.sh with the latest code, api/openapi-spec/swagger.json file should not be changed when no changes were made in the code.

When I tried the same thing a few week back, I didn't encounter this problem.

**How to reproduce it (as minimally and precisely as possible)**:
Download the latest kubernetes/kubernetes code and run the following script to update openapi spec.

```
# ./hack/update-openapi-spec.sh

make: Entering directory '/root/Github/kubernetes'
make[1]: Entering directory '/root/Github/kubernetes'
stat /root/Github/kubernetes/_output/local/Github/kubernetes/third_party/etcd-v3.3.15-linux-amd64/Documentation/README.md: no such file or directory
make[1]: Leaving directory '/root/Github/kubernetes'
+++ [1103 02:32:07] Building go targets for linux/amd64:
    cmd/kube-apiserver
make: Leaving directory '/root/Github/kubernetes'
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /tmp/tmp.EcaRTaU7fn --listen-client-urls http://127.0.0.1:2379 --log-level=debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.
+++ [1103 02:34:36] On try 1, etcd: : {"health":"true"}
{"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"2","raft_term":"2"}}+++ [1103 02:34:36] Starting kube-apiserver
+++ [1103 02:34:47] On try 7, apiserver: : ok
+++ [1103 02:34:47] Updating
    ./hack/../api/openapi-spec
+++ [1103 02:34:48] Clean up complete
```
api/openapi-spec/swagger.json became empty after this.

**Anything else we need to know?**:
There's a line in update-openapi-spec.sh that runs a curl command `curl -w "\n" -kfs --oauth2-bearer dummy_token  https://127.0.0.1:8050/openapi/v2`.

This line was changed a few days ago by this commit: https://github.com/kubernetes/kubernetes/commit/cfc2b330a7027e109ac26fc6a86b68898883099e
PR: https://github.com/kubernetes/kubernetes/pull/95856

I checked out the commit before cfc2b330a7027e109ac26fc6a86b68898883099e and ran /update-openapi-spec.sh.  It works.  swagger.json is not empty any more.

```
# git checkout eb63da77ea59af34033feb39bc8ccb5ce57c14a0
Note: switching to 'eb63da77ea59af34033feb39bc8ccb5ce57c14a0'.

# ./hack/update-openapi-spec.sh
make: Entering directory '/root/Github/kubernetes'
make[1]: Entering directory '/root/Github/kubernetes'
+++ [1103 02:49:46] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/conversion-gen
make[1]: Leaving directory '/root/Github/kubernetes'
+++ [1103 02:50:25] Building go targets for linux/amd64:
    cmd/kube-apiserver
make: Leaving directory '/root/Github/kubernetes'
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /tmp/tmp.fyS4an1E6o --listen-client-urls http://127.0.0.1:2379 --log-level=debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.
+++ [1103 02:51:47] On try 1, etcd: : {"health":"true"}
{"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"2","raft_term":"2"}}+++ [1103 02:51:47] Starting kube-apiserver
+++ [1103 02:51:51] On try 5, apiserver: : ok
+++ [1103 02:51:51] Updating
    ./hack/../api/openapi-spec
+++ [1103 02:51:51] SUCCESS
+++ [1103 02:51:53] Clean up complete
```

Is there any new setting required before running ./hack/update-openapi-spec.sh now?

**Environment**:
- Kubernetes version (use `kubectl version`):
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.0-beta.0.127+dd56138e55d11c-dirty", GitCommit:"dd56138e55d11c557bd5aebe5fcd4b68eb7c4191", GitTreeState:"dirty", BuildDate:"2020-11-01T02:55:09Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by knight42**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The flag `--oauth2-bearer` might have no effect in some environment.

**Which issue(s) this PR fixes**:

Fixes #96161

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by knight42** [comment]
/cc @liggitt 

Would you mind taking a look?

File: hack/lib/util.sh [hunk 1]
```diff
@@ -67,15 +65,25 @@ kube::util::wait_for_url() {
...
+
+kube::util::wait_for_url_with_bearer_token() {
+  local url=${1}
+  local token=${2}
   local prefix=${3:-}
   local wait=${4:-1}
   local times=${5:-30}
   local maxtime=${6:-1}
 
-  kube::util::wait_for_url "${url}" "${prefix}" "${wait}" "${times}" "${maxtime}" "--oauth2-bearer ${token}"
```
Comment (liggitt) [inline hack/lib/util.sh:78]: could this PR be simplified to just change this line to `-H \"Authorization: Bearer ${token}\"`?

Comment (knight42) [reply hack/lib/util.sh:78]: I have tried this before, but due to the word splitting mechanism in bash, curl will see `-H` `"Authorization:` `Bearer` `${token}"`, which is not expected.

Comment (liggitt) [reply hack/lib/util.sh:78]: what about using `"${@:6}"` instead of `extra_args`?

Comment (liggitt) [reply hack/lib/util.sh:78]: (I'm trying to avoid a third utility method, and minimize change, if possible)

Comment (knight42) [reply hack/lib/util.sh:78]: sound like a good idea, I just learn that we could do slicing in bash

**Pull request merged**